airflow/jobs/scheduler_job.py

```diff
@@ -764,6 +765,26 @@ def _execute(self) -> None:
                     self.log.exception("Exception when executing DagFileProcessorAgent.end")
             self.log.info("Exited execute loop")
 
+    def _update_dag_run_state_for_paused_dags(self):
+        try:
+            paused_dag_ids = DagModel.get_all_paused_dag_ids()
+            for dag_id in paused_dag_ids:
+                if dag_id in self._paused_dag_without_running_dagruns:
+                    continue
```

Comment:
dag_ids are never removed from this set. If I unpause the DAG after this has run and then later re-pause it, then this method skips it.

Reply:
good catch. let me open a PR.